python/rmm/tests/test_rmm.py

```diff
@@ -887,3 +888,29 @@ def func_with_arg(x):
     rmm.unregister_reinitialize_hook(func_with_arg)
     rmm.reinitialize()
     assert L == [2]
+
+
+@pytest.mark.parametrize(
+    "cuda_ary",
+    [
+        lambda: rmm.DeviceBuffer.to_device(b"abc"),
+        lambda: cuda.to_device(np.array([97, 98, 99, 0, 0], dtype="u1")),
+    ],
+)
+@pytest.mark.parametrize(
+    "make_copy", [lambda db: db.copy(), lambda db: copy.deepcopy(db)]
```

Comment:
Maybe test `copy.copy` rather than the deep-copy fallback to `__copy__`?
```suggestion
    "make_copy", [lambda db: db.copy(), lambda db: copy.copy(db)]
```